### PR TITLE
[JS/Web] Added CumSum operator to JSEP

### PR DESCRIPTION
**Pull request by satyajandhyala**
### Description
Added CumSum operator

### Motivation and Context
Reduce CPU <->GPU data movement.

### CODE REVIEW

**Comment by guschmue** [comment]
taking it for test

File: js/web/lib/wasm/jsep/webgpu/ops/cumsum.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              (inputShape.length === 1 ? 'i32(input_shape)' : 'i32(input_shape[uniforms.axis])');
+          let lowerLimit = attributes.reverse === 1 ? index : '0';
+          let upperLimit = attributes.reverse === 1 ? max : index + ' + 1';
+          if (attributes.exclusive === 1) {
+            if (attributes.reverse === 1) {
+              lowerLimit = lowerLimit + ' + 1';
+            } else {
+              upperLimit = upperLimit + ' - 1';
+            }
+          }
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/webgpu/ops/cumsum.ts:46]: Please move the logic of `index`, `max`, `lowerLimit` and `upperLimit` into `getShaderSource` to simplify the body of `createCumsumProgramInfo`.

File: js/web/lib/wasm/jsep/webgpu/ops/cumsum.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import {ShapeUtil} from '../../util';
+import {AttributeWithCacheKey, createAttributeWithCacheKey} from '../attribute-with-cache-key';
+import {ComputeContext, ProgramInfo} from '../types';
+
+import {createTensorShapeVariables, inputVariable, outputVariable, ShaderHelper} from './common';
+
+
+export interface CumSumAttributes extends AttributeWithCacheKey {
+  readonly exclusive: number;
+  readonly reverse: number;
```
Comment (fs-eire) [inline js/web/lib/wasm/jsep/webgpu/ops/cumsum.ts:15]: use boolean

Comment (satyajandhyala) [reply js/web/lib/wasm/jsep/webgpu/ops/cumsum.ts:15]: Yeah. Will do. I thought about this at time of coding but I was not sure whether it makes any difference.

**Review by guschmue — APPROVED** [review]

**Pull request merged**